src/main/java/nl/tudelft/ti2206/group9/gui/ExternalTicker.java

```diff
@@ -97,20 +98,78 @@ private Group renderEntities() {
 				entityBox.setTranslateZ(entity.getCenter().getZ());
 
 				if (entity instanceof Player) {
-					entityBox.setMaterial(new PhongMaterial(Color.ORANGE));
+					entityBox.setMaterial(Style.PLAYER);
 				} else if (entity instanceof Coin) {
-					entityBox.setMaterial(new PhongMaterial(Color.GOLD));
+					entityBox.setMaterial(Style.COIN);
 				} else if (entity instanceof Log) {
-					entityBox.setMaterial(new PhongMaterial(Color.BROWN));
+					entityBox.setMaterial(Style.WOOD);
 				} else if (entity instanceof Pillar) {
-					entityBox.setMaterial(new PhongMaterial(Color.WHITE));
-				} else {
-					entityBox.setMaterial(new PhongMaterial(Color.GRAY));
+					entityBox.setMaterial(Style.PILLAR);
+				} else /*if (entity instanceof Fence) */ {
+					entityBox.setMaterial(Style.FENCE);
 				}
+
+				entityBox.setCache(true);
+				entityBox.setCacheHint(CacheHint.SPEED);
 				entities.getChildren().add(entityBox);
             }
 		}
 		return entities;
 	}
+
+	/**
+	 * This method is for producing track pieces.
+	 * This is a working version, however refactoring is needed.
+	 * @return Group with all the track parts
+	 */
+	private Group renderTrack() {
+		Group result = new Group();
+		for (int i = 0; i < 100; i++) {
+			for (int j = 0; j < 3; j++) {
+				final Box trackPiece = new Box(1.5, 0, 1.5);
+				trackPiece.setTranslateX(j - 1);
+				trackPiece.setTranslateZ(i);
+				trackPiece.setMaterial(Style.FLOOR);
+				trackPiece.setCache(true);
+				trackPiece.setCacheHint(CacheHint.SPEED);
+				result.getChildren().add(trackPiece);
+			}
+		}
+		return result;
+	}
+
+	/**
+	 * The same applies here as described above.
+	 * This part handles the walls.
+	 * h: 2 walls
+	 * i: 500 wallpieces in depth
+	 * j: 3 wallpieces in height
+	 * @return Group with all the wallpieces
+	 */
+	private Group renderWall() {
+		Group result = new Group();
+		for (int h = 0; h < 2; h++) {
+			for (int i = 0; i < 100; i++) {
+				for (int j = 0; j < 3; j++) {
+					double random = Math.random();
+					final Box wallPiece = new Box(0, 1, 1);
+					wallPiece.setTranslateY(j - 3);
+					wallPiece.setTranslateZ(i);
+					wallPiece.setTranslateX(h * 3 - 1.5);
+					if (random < 0.5) {
+						wallPiece.setMaterial(Style.BRICK);
+					} else if (random > 0.5 && random < 0.75) {
```

Comment:
In this part of the code are many magic numbers. I suggest replacing the for-loop boundaries by `Track.LENGTH` and `Track.WIDTH`, and for the other numbers you'll have to create constants.
<layout,logic